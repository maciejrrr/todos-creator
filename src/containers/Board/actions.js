import { ADD_BOARD } from './constants';

export const addBoard = ({ board }) => ({
  type: ADD_BOARD,
  board,
});
