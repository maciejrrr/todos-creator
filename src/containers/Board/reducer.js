import { ADD_BOARD } from './constants';

const initialState = {
  boards: [],
};

const addBoardState = (state, action) => ({
  ...state,
  boards: [...state.boards, action.board],
});

const boardsReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_BOARD:
      return addBoardState(state, action);
    default:
      return state;
  }
};

export default boardsReducer;
