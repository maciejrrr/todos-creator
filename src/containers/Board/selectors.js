import { createSelector } from 'reselect';

export const boardsStateSelector = state => state.boardsState;

export const allBoardsSelector = createSelector(
  boardsStateSelector,
  boardsState => boardsState.boards
);
