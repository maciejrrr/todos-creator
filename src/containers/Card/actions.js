import { ADD_CARD, EDIT_CARD, ADD_TASK } from './constants';

export const addCard = ({ card }) => ({
  type: ADD_CARD,
  card,
});

export const editCard = ({ cardId, name }) => ({
  type: EDIT_CARD,
  cardId,
  name,
});

export const addTask = ({ task }) => ({
  type: ADD_TASK,
  task,
});
