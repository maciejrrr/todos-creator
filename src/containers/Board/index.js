import React, { Component } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import styled from 'styled-components';

import AddCard from '../../components/AddCard';
import Card from '../Card';

import { cardsSelector } from './selectors';
import { addCard } from './actions';

const CardsContainer = styled.div`
  display: flex;
  flex-direction: row;
`;

export class Board extends Component {
  renderCards() {
    const { cards } = this.props;
    return cards.map(card => <Card key={card.id} card={card} />);
  }

  render() {
    const { addCard } = this.props;
    return (
      <CardsContainer>
        {this.renderCards()}
        <AddCard addCard={addCard} />
      </CardsContainer>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  cards: cardsSelector,
});

const mapDispatchToprops = dispatch => ({
  addCard: ({ card }) => dispatch(addCard({ card })),
});

export default connect(mapStateToProps, mapDispatchToprops)(Board);
