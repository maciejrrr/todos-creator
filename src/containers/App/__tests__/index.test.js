import { App, mapDispatchToProps } from '../index';
import { addBoard } from '../../Board/actions';

jest.mock('../../Board', () => 'Board');

describe('App component', () => {
  let props;
  beforeEach(() => {
    props = {
      boards: [],
      addBoard: jest.fn(),
    };
  });

  describe('snapshots', () => {
    describe('default state', () => {
      it('renders Board container without boardId', () => {
        const tree = renderer.create(<App {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('with boardId in state', () => {
      it('renders Board container with boardId', () => {
        const container = renderer.create(<App {...props} boards={[{ id: 1, name: 'test' }]} />);
        container.getInstance().handleBoardChange(1);
        expect(container.toJSON()).toMatchSnapshot();
      });
    });
  });

  describe('handleBoardChange', () => {
    it('sets boardId state', () => {
      const value = 2;
      const component = shallow(<App {...props} />);
      component.instance().handleBoardChange(value);
      expect(component.state().boardId).toBe(value);
    });
  });

  describe('handleAddBoard', () => {
    let item;
    let handleBoardChangeMock;
    let component;
    beforeEach(() => {
      item = { id: 2, name: 'test board' };
      handleBoardChangeMock = jest.fn();
      component = shallow(<App {...props} />);
      component.instance().handleBoardChange = handleBoardChangeMock;
      component.instance().handleAddBoard({ item });
    });

    it('calls addBoard', () => {
      expect(props.addBoard).toHaveBeenCalledWith({ board: item });
    });

    it('calls handleBoardChange', () => {
      expect(handleBoardChangeMock).toHaveBeenCalledWith(item.id);
    });
  });

  describe('mapDispatchToprops', () => {
    let dispatch;
    let result;
    beforeEach(() => {
      dispatch = jest.fn();
      result = mapDispatchToProps(dispatch);
    });

    describe('addBoard', () => {
      it('dispatches addBoard with card object', () => {
        const board = { id: 1, name: 'test board' };
        result.addBoard({ board });
        expect(dispatch).toBeCalledWith(addBoard({ board }));
      });
    });
  });
});
