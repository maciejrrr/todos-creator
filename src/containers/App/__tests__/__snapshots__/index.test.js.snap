// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshots default state renders Board container without boardId 1`] = `
<div
  className="sc-gzVnrw eSJPPX ant-layout"
>
  <div
    className="sc-htoDjs hRouSj ant-layout-header"
  >
    <div
      className="sc-EHOje heLVgc"
    >
      <div
        className="sc-bZQynM fMhQSn ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          aria-autocomplete="list"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
                    ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="Select board"
            >
              Select board
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="unselectable"
          >
            <b />
          </span>
        </div>
      </div>
      <form
        className="sc-bxivhb iGVlRn"
        onSubmit={[Function]}
      >
        <input
          className="sc-ifAKCX cMqBtW"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Add a board"
          value=""
        />
      </form>
    </div>
  </div>
  <Board
    boardId={null}
  />
</div>
`;

exports[`App component snapshots with boardId in state renders Board container with boardId 1`] = `
<div
  className="sc-gzVnrw eSJPPX ant-layout"
>
  <div
    className="sc-htoDjs hRouSj ant-layout-header"
  >
    <div
      className="sc-EHOje heLVgc"
    >
      <div
        className="sc-bZQynM fMhQSn ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          aria-autocomplete="list"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
                    ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="test"
            >
              test
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="unselectable"
          >
            <b />
          </span>
        </div>
      </div>
      <form
        className="sc-bxivhb iGVlRn"
        onSubmit={[Function]}
      >
        <input
          className="sc-ifAKCX cMqBtW"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Add a board"
          value=""
        />
      </form>
    </div>
  </div>
  <Board
    boardId={1}
  />
</div>
`;
